### PR TITLE
Update finetune and oneshot tests

### PR DESCRIPTION
**Pull request by dsikka**
SUMMARY:
- Properly update finetune tests such that they can run on a nightly cadence. This includes defining a simple config file that can be used by the tests which defines the cadence 
- Mark finetune tests which are not `integration` tests as `unit`
- Seems as oneshot does not work with auto - update the tests to use `cuda: 0`

### CODE REVIEW

**Review by kylesayrs — APPROVED** [review]
Nice

File: .github/workflows/test-check.yaml [hunk 1]
```diff
@@ -114,10 +114,14 @@ jobs:
       - name: "⚙️ Install dependencies"
         id: install
         run: pip3 install .[dev]
-      - name: "🔬 Running transformers tests[skipping over finetune]"
+      - name: "🔬 Running transformers tests"
         if: always() && steps.install.outcome == 'success'
         run: |
           pytest tests/llmcompressor/transformers/compression -v
```
Comment (Satrat) [inline .github/workflows/test-check.yaml:120]: What about the tests under obcq/gptq/oneshot/sparsification?

File: tests/llmcompressor/transformers/obcq/obcq_configs/completion/gpu/llama_7b_quant.yaml [hunk 2]
```diff
@@ -3,6 +3,6 @@ test_type: "regression"
 model: "meta-llama/Llama-2-7b-hf"
 dataset: open_platypus
 recipe: "tests/llmcompressor/transformers/obcq/recipes/quant.yaml"
-device: "auto"
+device: "cuda:0"
```
Comment (Satrat) [inline tests/llmcompressor/transformers/obcq/obcq_configs/completion/gpu/llama_7b_quant.yaml:6]: What was the error you were getting with auto? I think we want at least one test confirming that auto is functional

Comment (dsikka) [reply tests/llmcompressor/transformers/obcq/obcq_configs/completion/gpu/llama_7b_quant.yaml:6]: "Unspecified CUDA launch errors" - this was on an a100

File: .github/workflows/test-check.yaml (hunk 1 above)
Comment (dsikka) [reply .github/workflows/test-check.yaml:120]: They're all running. Lines 117 onwards. Just separating them out into separate steps

**Pull request merged**